cmd/utils/flags.go

```diff
@@ -7,10 +7,40 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"path"
+	"strings"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
+const (
+	NodeKey              = "node"
+	SubscriptionIDKey    = "subscription"
+	NodeResourceGroupKey = "node-resource-group"
+	VMSSKey              = "vmss"
+	InstanceIDKey        = "instance-id"
+	ResourceIDKey        = "id"
+)
+
+// We need package level variables to ensure that the viper flag binding works correctly.
+// See: https://github.com/spf13/viper/issues/375#issuecomment-578552586
+var (
+	node              string
+	subscriptionID    string
+	nodeResourceGroup string
+	vmScaleSet        string
+	instanceID        string
+	resourceID        string
+)
+
+func InitFlags() {
+	viper.AutomaticEnv()
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.SetConfigFile(ConfigFile())
+	viper.ReadInConfig()
```

Comment:
Could you please describe here what is the configuration we are setting, e.g. priority: flag > env > config-file, and so on?